### PR TITLE
Make traits.examples into a package

### LINKED ISSUE #1302: Make traits.examples into a package
There's a new module `traits._entry_point`. The purpose of this file is to make the examples available to the `etsdemo` application, but that's not obvious from the location or the name of the file. I propose that we:

- make `traits.examples` into a package (i.e., add an `__init__.py` file)
- move the code in `traits._entry_point` into the `traits.examples` package, and update the setup script accordingly
- rename `_entry_point` to something more meaningful

I'd also propose that the `pkg_resources` import be made local to the function in `_entry_point.py`, so that Traits itself remains free of external dependencies.

### PR DESCRIPTION
**Pull request by aaronayres35**
fixes #1302 

In this PR, an `__init__.py` file is added to `traits.examples` to make it a package, `_entry_point.py` is renamed `_etsdemo_info.py` and moved into the package with the `setup.py` entry point updated accordingly, and a very simple test is added for the `introduction` function in `_etsdemo_info.py` (I wasn't exactly sure what all we would want to test here). Also the `pkg_resource` import is moved inside the `introduction` function as mentioned on the issue.

I confirmed that I could launch the etsdemo application and still see the contributed examples.

**Checklist**
- [x] Tests
~- [ ] Update API reference (`docs/source/traits_api_reference`)~
~- [ ] Update User manual (`docs/source/traits_user_manual`)~
~- [ ] Update type annotation hints in `traits-stubs`~

### CODE REVIEW

File: traits/examples/tests/test_etsdemo_info.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+import os
+import unittest
+
+from traits.examples._etsdemo_info import introduction
+from traits.testing.optional_dependencies import (
+    pkg_resources,
```
Comment (mdickinson) [inline traits/examples/tests/test_etsdemo_info.py:15]: Unused import?

**Review by mdickinson — APPROVED** [review]
LGTM, modulo an unused import.

**Comment by mdickinson** [comment]
I've changed the branch protections to not require Travis CI (but those changes aren't backed up in terraform).

Closing and re-opening to retrigger CI.

**Comment by mdickinson** [comment]
Slack discussion says that Travis CI should be working again. Closing and re-opening to test that statement.

**Comment by mdickinson** [comment]
@kitchoi Okay to merge this once CI signs off?

**Comment by mdickinson** [comment]
> Slack discussion says that Travis CI should be working again. Closing and re-opening to test that statement.

And the test failed. But that shouldn't prevent merging this PR.

**Review by kitchoi — APPROVED** [review]
LGTM too.

Looks like Travis CI + MacOS + OSS is having some build issues recently.

**Pull request merged**